pymc3/sampling.py

```diff
@@ -1073,12 +1083,20 @@ def sample_ppc_w(traces, samples=None, models=None, weights=None,
     trace = []
     for i, j in enumerate(n):
         tr = traces[i]
-        len_trace = len(tr) 
-        nchain = tr.nchains 
+        len_trace = len(tr)
+        try:
+            nchain = tr.nchains
```

Comment:
Why the try/except? `tr.nchains` works even for traces with only 1 chain, right?

Reply:
Not work if you pass a list, it doesnt have `.nchains`. For example, you can do:
```python
with model:
    map1 = pm.find_MAP()
ppc = pm.sample_ppc([map1], sample=100, model=model)
```

Reply:
but `sample_ppc_w` is always expecting a list of traces

Reply:
yeah I was thinking the same, but it is a bit of a corner case as `sample_ppc` allows a list, and sample_ppc_w should as well, something like:
sample_ppc_w([[map1], [map2]], [model1, model2])
what do you think?

Reply:
You are right, a general solution is a better option.
Could you add a line in the doctring of `sample_ppc_w`, something similar as what you did for `sample_pcc`?